activerecord/lib/active_record/migration/compatibility.rb

```diff
@@ -178,6 +178,15 @@ def add_column(table_name, column_name, type, **options)
           super
         end
 
+        def change_column(table_name, column_name, type, **options)
+          if type == :datetime
+            options[:precision] ||= nil
+          end
+
+          type = PostgreSQLCompat.compatible_timestamp_type(type, connection)
+          super
```

Comment:
What do you think about extracting this logic into a helper method that can be shared between the `add_column' and `change_column' methods?

Reply:
Thanks for the review! I do think there's an opportunity for a refactor here, but I'm not 100% sure what it should be yet (maybe we want to use this new method in all of the `V6_1` column methods?). Going through these tests it appears that coverage isn't great so I think I would prefer to fix those issues first and ensure everything is working as expected before refactoring. How does that sound?

Reply:
That makes sense.